org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/SyntheticFilter.java

```diff
@@ -35,6 +35,11 @@ public void filter(final MethodNode methodNode,
 				return;
 			}
 
+			if (KotlinDefaultArgumentsFilter
```

Comment:
Side note: When I see this I more and more like @Godin 's idea to separate filters for different compilers.

Reply:
@marchof don't know about whom you're talking 😆

Not sure that separation can remove this dependency between filters, i.e. not every but some `synthetic` methods anyway should be filtered in Kotlin.